framework/src/main/java/org/checkerframework/framework/source/SourceChecker.java

```diff
@@ -3566,5 +3552,36 @@ public String toString() {
           + source
           + '}';
     }
+
+    /**
+     * Compares {@code other} with {@code this} {@link CheckerMessage}. Compares first by position
+     * at which the error will be printed, then by kind of message, then the order in which the
+     * checkers run, and finally by the message string.
+     *
+     * @param other the other CheckerMessage
+     * @return a negative integer, zero, or a positive integer if this CheckerMessage is less than,
+     *     equal to, or greater than {@code other}
+     */
+    @Override
+    public int compareTo(CheckerMessage other) {
+      int byPos = InternalUtils.compareDiagnosticPosition(this.source, other.source);
+      if (byPos != 0) {
+        return byPos;
+      }
+
+      int kind = this.kind.compareTo(other.kind);
+      if (kind != 0) {
+        return kind;
+      }
+
+      // Sort by order in which the checkers are run. (All the subcheckers,
+      // followed by the checker.)
+      int thisIndex = this.checker.getSubCheckerIndex();
+      int otherIndex = other.checker.getSubCheckerIndex();
+      if (thisIndex == otherIndex) {
+        return this.message.compareTo(other.message);
+      }
+      return Integer.compare(thisIndex, otherIndex);
```

Comment:
The comment says "finally by message string", but this final comparison isn't of message strings.